Rules/CommonScripts/ChatCommands.as

```diff
@@ -313,3 +364,25 @@ bool onClientProcessChat(CRules@ this, const string& in text_in, string& out tex
 
 	return true;
 }
+
+
+void onCommand(CRules@ this, u8 cmd, CBitStream @para)
+{
+	if (cmd == this.getCommandID("SendChatMessage"))
+	{
+		string errorMessage = para.read_string();
+		SColor col = SColor(para.read_u8(), para.read_u8(), para.read_u8(), para.read_u8());
+		client_AddToChat(errorMessage, col);
+	}
+}
+
+bool IsBlacklisted(string name)
+{
+	return  name=="hall" || // used to dig hole to bottom of the map, spawns lots of migrants
+			name=="shark" || // greif
+			name=="bison" ||
+			name=="necromancer" || // annoying / grief
+			name=="greg" || // annoying / grief
+			name=="ctf_flag" || // sound spam
+			name=="flag_base";// sound spam / grief
```

Comment:
You should put all these names in an array at the top of the file so it's neater.